wallet/src/main/java/com/android/identity_credential/wallet/ui/destination/provisioncredential/EvidenceRequest.kt

```diff
@@ -793,6 +1127,137 @@ fun <ResultT> EvidenceRequestIcaoView(
+        targetValue = 0f,
+        animationSpec = infiniteRepeatable(
+            animation = tween(durationMillis = transitionDurationMs, easing = LinearEasing),
+            repeatMode = RepeatMode.Restart
+        ), label = "alpha animation"
+    )
+    val tintColorThemed = if (!isSystemInDarkTheme()) {
+        colorResource(id = R.color.contactless_blue)
+    } else {
+        colorResource(id = R.color.contactless_blue).inverse()
+    }
+
+    // start composing the "nfc heartbeat" animation
+    Box(
+        contentAlignment = Alignment.Center,
+    ) {
+        // don't show the radius animation when encountering an error
+        if (nfcAnimationStatus != NfcAnimationStatus.Error) {
+            Box(
+                modifier = Modifier
+                    .size(iconSize) // heartbeat size
+                    .graphicsLayer(scaleX = scale, scaleY = scale)
+                    .alpha(alpha)
+                    .background(tintColorThemed, CircleShape)
+            )
+        }
+        // Clip the icon to be circular
+        Box(
+            modifier = Modifier.clip(CircleShape)
+        ) {
+            // get the main and background colors of the "contactless" icon
+            val (tintColor, bgColor) = when (nfcAnimationStatus) {
+                // waiting to begin reading an NFC document
+                // contactless blue color icon with the same color for background having a low alpha
+                NfcAnimationStatus.Initial -> Pair(
+                    tintColorThemed,
+                    tintColorThemed.copy(alpha = animationBgAlpha)
+                )
+                // once an NFC document is detected, inverse colors of "Initial" status
+                NfcAnimationStatus.Connected -> Pair(
+                    MaterialTheme.colorScheme.background,
+                    tintColorThemed
+                )
+                // Error while initializing/reading NFC card
+                // white icon with light red background with a lower alpha
+                NfcAnimationStatus.Error -> Pair(
+                    MaterialTheme.colorScheme.background,
+                    Color.Red.copy(
+                        alpha =
+                        if (isSystemInDarkTheme()) {
+                            1F
+                        } else {
+                            0.3F
+                        }
+                    )
+                )
+                // NFC scan complete -- colors are not used b/c icon is changed entirely
+                else -> Pair(Color.Transparent, Color.Transparent)
+            }
+            // compose the "contactless" icon
+            Icon(
+                painter = painterResource(id = R.drawable.contactless),
+                contentDescription = stringResource(
+                    R.string.accessibility_artwork_for,
+                    stringResource(id = R.string.evidence_nfc_scan_title)
+                ),
+                // icon color
```

Comment:
nit: Comments that just repeat what the next line is doing when it's already pretty clear just clutter up the code. See go/tott/563.

This one seems unnecessary. Maybe also some of the other comments around this.